src/main/java/com/microsoft/azure/vmagent/AzureVMAgentTemplate.java

```diff
@@ -613,6 +662,33 @@ public ListBoxModel doFillAgentLaunchMethodItems() {
             return model;
         }
 
+        public ListBoxModel doFillExistStorageAccountNameItems(
+                @RelativePath("..") @QueryParameter final String azureCredentialsId,
+                @RelativePath("..") @QueryParameter final String storageAccountNameReferenceType,
+                @RelativePath("..") @QueryParameter final String newResourceGroupName,
+                @RelativePath("..") @QueryParameter final String existResourceGroupName,
+                @QueryParameter final String storageAccountType) throws IOException, ServletException {
+            ListBoxModel model = new ListBoxModel();
+
+            try {
+                AzureCredentials.ServicePrincipal servicePrincipal = AzureCredentials.getServicePrincipal(azureCredentialsId);
```

Comment:
If user doesn't select any service principle, I guess the following code will throw. Though we have try catch here, let's also add some defense code here to avoid unnecessary exception logs.